### PR TITLE
Support addressShuffleMode to Spring Amqp Autoconfiguration

### PR DESCRIPTION
**Pull request by 01045972746**
This support connecting to RabbitMQ clustered host addresses with shuffling.
https://docs.spring.io/spring-amqp/reference/html/#cluster

Because of `shuffle-addresses` mode is deprecated, i have added `addressShuffleMode` property which is `String`.
Please review and comment. Thanks.

### CODE REVIEW

File: spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/amqp/RabbitProperties.java [hunk 1]
```diff
@@ -87,6 +89,11 @@
 	 */
 	private String addresses;
 
+	/**
+	 * Shuffling mode for connecting host. Default mode is NONE.
+	 */
+	private String addressShuffleMode = "NONE";
```
Comment (snicoll) [inline spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/amqp/RabbitProperties.java:95]: The enum should be exposed rather as it provides better metadata

File: spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/amqp/RabbitProperties.java [hunk 2]
```diff
@@ -87,6 +89,11 @@
 	 */
 	private String addresses;
 
+	/**
+	 * Shuffling mode for connecting host. Default mode is NONE.
```
Comment (snicoll) [inline spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/amqp/RabbitProperties.java:93]: The default value should not be documented as it's the role of the configuration metadata to do that (look at other keys for inspiration).

File: spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/amqp/RabbitProperties.java [hunk 3]
```diff
@@ -282,6 +289,28 @@ public void setVirtualHost(String virtualHost) {
...
+	}
+
+	/**
+	 * If addresses have been set and address shuffle mode has been set it is returned.
+	 * Otherwise returns the result of calling {@code getAddressShuffleMode()}.
+	 * @return the address shuffle mode
+	 * @see #setAddressShuffleMode(String)
+	 * @see #getAddressShuffleMode()
+	 */
+	public AbstractConnectionFactory.AddressShuffleMode determineAddressShuffleMode() {
```
Comment (snicoll) [inline spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/amqp/RabbitProperties.java:303]: I don't think we should do that. Rather we should set whatever has been set by the user.

**Review by snicoll — CHANGES_REQUESTED** [review]
Thanks for the suggestion and the PR. I've made a few comments, can you please have a look to those and also add the necessary tests to exercise the code that you've changed?

**Comment by 01045972746** [comment]
@snicoll , i have edited what you mention. please re-review plz.

**Comment by snicoll** [comment]
@01045972746 thank you for making your first contribution to Spring Boot. I've polished your contribution and added a test.

**Pull request closed without merging**